### PR TITLE
[Refactor][Delete][MRV2]Patch upstream mrv2 apply_top_k_top_p with current triton implementation

### PR DESCRIPTION
**Pull request by AuroraEmiya**
### What this PR does / why we need it?
Revert previous triton patch on ops apply_top_k_top_p. And patch it with implementation from v1 sample
### Does this PR introduce _any_ user-facing change?
None
### How was this patch tested?
this patch would be tested by test_basic.py

- vLLM main: https://github.com/vllm-project/vllm/commit/ba07e4a48fc951300d97eb506217dd530583dea3

### CODE REVIEW

File: vllm_ascend/worker/v2/sample/apply_top_k_top_p.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import torch
+import torch_npu  # noqa: F401
+
+
+def apply_top_k_top_p(logits: torch.Tensor, k: torch.Tensor | None, p: torch.Tensor | None) -> torch.Tensor:
+    if k is None and p is None:
+        return logits
+    # use cann ops
```
Comment (Mutian1207) [inline vllm_ascend/worker/v2/sample/apply_top_k_top_p.py]: will be reverted/removed when Sampler get patched in the future, add some comments

**Comment by Liamup777** [comment]
topktopp patch will be reverted/removed when Sampler gets patched in the future; pls add some comments

**Comment by Liamup777** [comment]
triton kernel latency:
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/l00914336/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/l00914336/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">



</head>

<body link="#467886" vlink="#96607D">


Device_id | OP Type | Core Type | Count | Total Time(us) | Min Time(us) | Avg Time(us) | Max Time(us) | Ratio(%)
-- | -- | -- | -- | -- | -- | -- | -- | --
10 | _topk_topp_kernel | AI_VECTOR_CORE | 51 | 757379.6 | 4344.28 | 14850.58 | 40799.68 | 6.864



</body>

</html>


apply_top_k_top_p with cann:
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/l00914336/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/l00914336/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">



</head>

<body link="#467886" vlink="#96607D">


Device_id | OP Type | Core Type | Count | Total Time(us) | Min Time(us) | Avg Time(us) | Max Time(us) | Ratio(%)
-- | -- | -- | -- | -- | -- | -- | -- | --
10 | ApplyTopKTopPWithSorted | MIX_AIV | 54 | 246671.2 | 2461.429 | 4567.985 | 4993.479 | 2.341



</body>

</html>

**Comment by Liamup777** [comment]
image

File: vllm_ascend/patch/worker/patch_v2/patch_triton.py [hunk 2]
```diff
@@ -41,9 +40,11 @@
 rejection_sampler_utils.rejection_sample = npu_rejection_sample
 rejection_sampler.rejection_sample = npu_rejection_sample
 dflash_speculator._prepare_dflash_inputs_kernel = _prepare_dflash_inputs_kernel_ascend
+# This patch may be revisited or reverted once the sampler layer modulizeation is done.
+sampler.apply_top_k_top_p = apply_top_k_top_p
+states.apply_top_k_top_p = apply_top_k_top_p
```
Comment (zouzy5137) [inline vllm_ascend/patch/worker/patch_v2/patch_triton.py]: Check if rejection_sampler needs to do this patch.

File: vllm_ascend/worker/v2/sample/apply_top_k_top_p.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import torch
+import torch_npu  # noqa: F401
+
+
+def apply_top_k_top_p(logits: torch.Tensor, k: torch.Tensor | None, p: torch.Tensor | None) -> torch.Tensor:
+    if k is None and p is None:
+        return logits
+    # use cann ops
+    return torch_npu.npu_top_k_top_p(logits, p, k)
```
Comment (zouzy5137) [inline vllm_ascend/worker/v2/sample/apply_top_k_top_p.py]: During warm-up stage, if k and p is None, this op will be skipped and kvcache calculation is not correct. Add a TODO or NOTE for this problem.

**Review by Ronald1995 — APPROVED** [review]
LGTM

**Review by chenlongxiao — APPROVED** [review]

File: vllm_ascend/patch/worker/patch_v2/patch_triton.py (hunk 2 above)
Comment (AuroraEmiya) [reply vllm_ascend/patch/worker/patch_v2/patch_triton.py]: There is no need for rejection_sampler to do this patch.
vllm.v1.work.gpu.spec_decode.rejection_sampler would call apply_top_k_top_p inside
RejectionSampler._verify by calling apply_sampling_params which is imported from states.

File: vllm_ascend/worker/v2/sample/apply_top_k_top_p.py (hunk 3 above)
Comment (AuroraEmiya) [reply vllm_ascend/worker/v2/sample/apply_top_k_top_p.py]: Done

**Review by wangxiyuan — DISMISSED** [review]

**Review by freyfwt — APPROVED** [review]
后续在上游把采样插件化，把这块patch消掉。

**Review by zouzy5137 — APPROVED** [review]

**Review by chenlongxiao — APPROVED** [review]

**Review by zzzzwwjj — DISMISSED** [review]

**Comment by AuroraEmiya** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Review by ningjingbengxiaohai — APPROVED** [review]
